### PR TITLE
Migrate to idiomatic error handling in app/slackimport.go

### LINKED ISSUE #9686: [Help Wanted] [MM-12610] Migrate to idiomatic error handling in the file `app/slackimport.go` in the mattermost-server repo.
If you're interested please comment here and come [join our "Contributors" community channel](https://pre-release.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://pre-release.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-12610)

This ticket is best suited to someone who is already familiar with what the "idiomatic error handling" migration is about. You can discuss this with us on the Mattermost server linked above, and see examples of these changes at commits such as: [https://github.com/mattermost/mattermost-server/commit/a6fa2b72d1375e554cc6f2e5ca2fd4884a8509f4]

Although suggests changing to , please don’t make these changes as part of this PR, as we have decided to keep the codebase consistent the way it is for the time being. At a future date, we will consider migrating the whole codebase.

### PR DESCRIPTION
**Pull request by SergeyShpak**
#### Summary
Migrate to idiomatic error handling in the file `app/slackimport.go` in the mattermost-server repo.

#### Ticket Link
Closes #9686

### CODE REVIEW

File: app/slackimport.go [hunk 1]
```diff
@@ -164,7 +161,8 @@ func (a *App) SlackAddUsers(teamId string, slackusers []SlackUser, importerLog *
 		password := model.NewId()
 
 		// Check for email conflict and use existing user if found
-		if result := <-a.Srv.Store.User().GetByEmail(email); result.Err == nil {
+		result = <-a.Srv.Store.User().GetByEmail(email)
```
Comment (grundleborg) [inline app/slackimport.go]: No need to split this one out, as the `result` variable isn't needed anywhere other than the `if` block, and there isn't any `else`.

Comment (SergeyShpak) [reply app/slackimport.go]: Makes perfect sense, thank you!

File: app/slackimport.go [hunk 2]
```diff
@@ -218,10 +214,9 @@ func (a *App) SlackAddBotUser(teamId string, log *bytes.Buffer) *model.User {
 	if mUser := a.OldImportUser(team, &botUser); mUser != nil {
 		log.WriteString(utils.T("api.slackimport.slack_add_bot_user.email_pwd", map[string]interface{}{"Email": botUser.Email, "Password": password}))
 		return mUser
-	} else {
-		log.WriteString(utils.T("api.slackimport.slack_add_bot_user.unable_import", map[string]interface{}{"Username": username}))
-		return nil
 	}
+	log.WriteString(utils.T("api.slackimport.slack_add_bot_user.unable_import", map[string]interface{}{"Username": username}))
```
Comment (lieut-data) [inline app/slackimport.go]: nit: Can you reverse this conditional, to check `mUser == nil` first and having the successful case follow?

File: app/slackimport.go [hunk 3]
```diff
@@ -659,7 +662,9 @@ func (a *App) SlackImport(fileData multipart.File, fileSize int64, teamID string
 				} else {
 					posts[channel] = append(posts[channel], newposts...)
 				}
-			} else if len(spl) == 3 && spl[0] == "__uploads" {
+				continue
+			}
+			if len(spl) == 3 && spl[0] == "__uploads" {
```
Comment (lieut-data) [inline app/slackimport.go]: I'd suggest not making this change: we aren't doing it on line 653 above, and this isn't really about error handling per se.

File: app/slackimport.go [hunk 4]
```diff
@@ -798,7 +803,9 @@ func (a *App) OldImportIncomingWebhookPost(post *model.Post, props model.StringI
 				if attachments, success := val.([]*model.SlackAttachment); success {
 					model.ParseSlackAttachment(post, attachments)
 				}
-			} else if key != "from_webhook" {
+				continue
+			}
+			if key != "from_webhook" {
```
Comment (lieut-data) [inline app/slackimport.go]: Same comment as earlier, but I don't think this change is necessary.

File: app/slackimport.go (hunk 3 above)
Comment (SergeyShpak) [reply app/slackimport.go]: I was not sure about this change; it seemed to make the code more readable, as there is another ```if``` enclosed just before. But I agree that it is a question of preference and that this kind of change does not belong in this commit.

File: app/slackimport.go (hunk 4 above)
Comment (SergeyShpak) [reply app/slackimport.go]: Agreed (see the comment before)

**Comment by SergeyShpak** [comment]
Thanks for the request @grundleborg. Could you check if the change I have made resolves the problem, please?

**Comment by grundleborg** [comment]
Just need to wait for the Github outage to finally end, and then I'll get this merged (probably tomorrow AM).

**Pull request merged**